src/main/java/org/sonarlint/intellij/config/project/SearchProjectKeyDialog.java

```diff
@@ -48,17 +48,19 @@
 
 public class SearchProjectKeyDialog extends DialogWrapper {
   private final Map<String, RemoteProject> projectsByKey;
+  private final boolean isSonarCloud;
 
   private String lastSelectedProjectKey;
   private JPanel contentPane;
   private JBScrollPane scrollPane;
   private JBList<RemoteProject> projectList;
 
-  public SearchProjectKeyDialog(Component parent, String selectedProjectKey, Map<String, RemoteProject> projectsByKey) {
+  public SearchProjectKeyDialog(Component parent, String selectedProjectKey, Map<String, RemoteProject> projectsByKey, boolean isSonarCloud) {
     super(parent, false);
     this.lastSelectedProjectKey = selectedProjectKey;
     this.projectsByKey = projectsByKey;
-    setTitle("Search Project in SonarQube Server");
+    this.isSonarCloud = isSonarCloud;
+    setTitle("Search Project in " + (isSonarCloud ? "SonarCloud" : "SonarQube"));
```

Comment:
I was wondering the same, but there are situations where we don't yet have a ServerConnection object (for example in the wizard). Still it might be interesting.